sources/models/src/lib.rs

```diff
@@ -77,6 +77,16 @@ struct ConfigurationFile {
     template_path: SingleLineString,
     #[serde(skip_serializing_if = "Option::is_none")]
     mode: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    overwrite_path_if_present: Option<bool>,
+}
+
+impl ConfigurationFile {
+    /// Checks whether the configuration file should be written if the path is present.
+    /// and overwrite_path_if_present is set.
+    pub fn should_overwrite_path_if_present(&self) -> bool {
+        self.overwrite_path_if_present != Some(false) || !Path::new(&self.path.as_ref()).exists()
```

Comment:
nit: calling this `should_render` might be more clear - it's kind of weird to return `true` from a function called `should_overwrite_path_if_present` when the path is not present